contracts/src/v0.8/dev/automation/2_1/KeeperRegistryLogicA2_1.sol

```diff
@@ -40,10 +40,25 @@ contract KeeperRegistryLogicA2_1 is
 
   uint8 public constant override upkeepVersion = UPKEEP_VERSION_BASE;
 
-  function checkUpkeep(
-    uint256 id
-  )
+  /**
+   * @dev this function will be deprecated in a future version of chainlink automation
+   */
```

Comment:
why so? I think it's valuable to have this for those upkeeps which have onchain checkData

Reply:
it will be passed explicitly by nodes in the future (feels simpler to have one endpoint regardless of upkeep type imho). Just an opinion for code hygiene, not a strong one tho.